### PR TITLE
Update example RBAC

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area auto-scaling
/kind enhancement

**What this PR does / why we need it**:
@plkokanov reported that there are missing RBAC rules for leader election and events in the example RBACs.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/8259

**Special notes for your reviewer**:
N/A

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by andrerun — APPROVED** [review]
Looks good.

This is now slightly behind latest thinking (since the changes we made together yesterday on the RBAC entities created by gardenlet - e.g. name-based restrictions). But I think we should get this change in, and then we can reiterate later, for complete convergence.

**Comment by ialidzhikov** [comment]
> This is now slightly behind latest thinking (since the changes we made together yesterday on the RBAC entities created by gardenlet - e.g. name-based restrictions). But I think we should get this change in, and then we can reiterate later, for complete convergence.

Thanks for bringing up this! I now had a look in the context of https://github.com/gardener/gardener-custom-metrics/pull/23 and I think we won't be able to introduce such a name-based restriction for the list/watch verbs. In https://github.com/gardener/gardener-custom-metrics/pull/23 we add a label selector to the controller-runtime's cache which I believe should result in list/watch requests by label selector. Hence, we cannot add a restriction for the resource names at least for list and watch. We can add the resource name based restriction for the get but then it doesn't make much sense conceptually because you are able to list all resources from all namespaces but able to get only 2 because of the resourceNames restriction.

The endpoints name restriction that is possible after https://github.com/gardener/gardener-custom-metrics/pull/22 is applied in this PR.

File: example/rbac.yaml [hunk 1]
```diff
@@ -4,127 +4,117 @@ metadata:
...
+  resources:
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - secrets
```
Comment (plkokanov) [inline example/rbac.yaml:86]: Can we move this up where the `pods` are also defined:
```
resources:
- pods
- secrets
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
One minor suggestion, rest looks good.

**Review by plkokanov — APPROVED** [review]
/lgtm

**Pull request merged**